packages/react-client/src/screenShareTrackManager.ts

```diff
@@ -0,0 +1,92 @@
+import { FishjamClient } from "@fishjam-dev/ts-client";
+import { useCallback, useEffect, useState } from "react";
+import { getRemoteTrack } from "./utils/tracks";
+
+const getTracks = (stream: MediaStream): { video: MediaStreamTrack; audio: MediaStreamTrack | null } => {
+  const video = stream.getVideoTracks()[0];
+  const audio = stream.getAudioTracks()[0] ?? null;
+
+  return { video, audio };
+};
+
+export const useScreenShare = <PeerMetadata, TrackMetadata>(tsClient: FishjamClient<PeerMetadata, TrackMetadata>) => {
+  const [stream, setStream] = useState<MediaStream | null>(null);
+  const [trackIds, setTrackIds] = useState<{ videoId: string; audioId?: string } | null>(null);
+
+  useEffect(() => {
+    if (!stream) return;
+    const { video, audio } = getTracks(stream);
+
+    const getTrackEndedHandler = (track: MediaStreamTrack) => () => {
+      stream.getTrackById(track.id)?.stop();
+    };
```

Comment:
Will this still work if we simplify it?

```suggestion
    const trackEndedHandler = (track: MediaStreamTrack) => {
      stream.getTrackById(track.id)?.stop();
    };
```

Or do we require a separate handler for audio and video?

Reply:
no :)